tests/test_common/test_cxtgeo_lowlevel.py

```diff
@@ -80,3 +80,70 @@ def test_well_mask_shoulder(depth, logseries, distance, expected):
     print(mask)
     assert res == 0
     assert (mask == expected).all()
+
+
+@pytest.mark.parametrize(
+    "xcoords, ycoords, zcoords, xval, yval, method, zexpected",
+    [
+        (
+            [0.0, 1.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0, 1.0],
+            [1.0, 2.0, 3.0, 4.0],
+            0.5,
+            0.5,
+            2,  # bilinear, nonrotated
+            2.5,
+        ),
+        (
+            [0.0, 1.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0, 1.0],
+            [1.0, 2.0, 3.0, 4.0],
+            0.0,
+            0.5,
+            2,  # bilinear, nonrotated
+            2.0,
+        ),
+        (
+            [0.0, 1.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0, 1.0],
+            [1.0, 2.0, 3.0, 4.0],
+            0.4999,
+            0.4999,
+            4,  # nearest node
+            1.0,
+        ),
+        (
+            [0.0, 1.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0, 1.0],
+            [1.0, 2.0, 3.0, 4.0],
+            0.50001,
+            0.49999,
+            4,  # nearest node
+            2.0,
+        ),
+        (
+            [0.0, 1.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0, 1.0],
+            [1.0, 2.0, 3.0, 4.0],
+            0.49999,
+            0.50001,
+            4,  # nearest node
```

Comment:
What is the nearest node value at the singularity (0.5, 0.5)?

Reply:
Would possibly be random? But good input I can have a look

Reply:
If random is accepted, you could say it has to be one of [1, 2, 3, 4].